server/src/main/java/org/opensearch/transport/InboundHandler.java

```diff
@@ -192,6 +217,41 @@ private void messageReceived(TcpChannel channel, InboundMessage message, long st
         }
     }
 
+    private void messageReceivedProtobuf(TcpChannel channel, NodeToNodeMessage message, long startTime) throws IOException {
```

Comment:
Could we please extract `ProtobufMessageHandler` and `NativeMessageHandler` (or any other name we think is good for current protocol), so the `InboundHandler` would be essentially just a router to the protocol handler ?

Reply:
Extracted this out.